src/chunk.ts

```diff
@@ -844,31 +854,21 @@ export class Chunk {
         skipBlockTo: SkipBlock,
         svgBlocks: any
     ) {
-        // Genesis block has a backward link to a random generated block, so that two
-        // identical genesis blocks have a different ID.
-        if (skipBlockTo.index != 0) {
-            // Iterate through all blocks
-            for (let i = 0; i < skipBlockTo.backlinks.length; i++) {
-                Utils.getBlock(
-                    skipBlockTo.backlinks[i], // Get all blocks that point to skipBlockTo
-                    this.roster
-                )
-                    .then((skipBlockFrom) => {
-                        this.appendArrows(
-                            xTranslate,
-                            skipBlockFrom,
-                            skipBlockTo,
-                            svgBlocks,
-                            i
-                        );
-                    })
-
-                    .catch((e) => {
-                        this.flash.display(
-                            Flash.flashType.ERROR,
-                            `Cannot find backward link: ${e}`
-                        );
-                    });
+        let index = skipBlockTo.index;
+        let mult = 1;
+        if (skipBlockTo.forwardLinks.length != 0) {
+            for (let i = 0; i < skipBlockTo.height; i++) {
+                if (index + mult <= this.lastAddedBlock.index) {
+                    // We do not draw arrows that point to non-existing blocks
+                    this.appendArrows(
+                        xTranslate,
+                        index + mult,
+                        skipBlockTo,
+                        svgBlocks,
+                        i
+                    );
+                }
+                mult *= skipBlockTo.baseHeight;
```

Comment:
There I would remove  `if (skipBlockTo.forwardLinks.length != 0) {` and add a check before `this.appendArrows` to check if the block has enough forward links, and print a warning if not:
```
if skipBlockTo.forwardLinks.length >= i { 
  this.appendArrows ...
} else { 
  flash.Warning(`missing forward link ${i} on block...`)
}
```

Columbus should show the state of the chain as close as possible to what is actually stored: doing the calculation instead of fetching blocks is nice from a performance perspective, but it should then not hide some potential problems that are in the chain, like missing forward links,

Reply:
Indeed, forward links would not be checked anymore without fetching the block. I adapted the code so that if there are less forward links than the height of the block the user is notified (taking into account that the links are not missing because they would point to a block that is not yet existent.)